src/flask/config.py

```diff
@@ -176,31 +176,48 @@ class and has ``@property`` attributes, it needs to be
             if key.isupper():
                 self[key] = getattr(obj, key)
 
-    def from_json(self, filename, silent=False):
-        """Updates the values in the config from a JSON file. This function
-        behaves as if the JSON object was a dictionary and passed to the
-        :meth:`from_mapping` function.
+    def from_file(self, filename, loads, silent=False):
+        """Update the values in the config from a file that is loaded using
+        the *loads* argument.  This method passes the loaded Mapping
+        to the :meth:`from_mapping` function.
 
         :param filename: the filename of the JSON file.  This can either be an
                          absolute filename or a filename relative to the
                          root path.
+        :param loads: a callable that loads the file text and returned a
+                      Mapping.
+        :type loads: Callable[[str], Mapping].
         :param silent: set to ``True`` if you want silent failure for missing
                        files.
 
-        .. versionadded:: 0.11
+        .. versionadded:: 1.2
         """
         filename = os.path.join(self.root_path, filename)
-
         try:
-            with open(filename) as json_file:
-                obj = json.loads(json_file.read())
+            with open(filename) as file_:
+                obj = loads(file_.read())
```

Comment:
Should this use `load(file_)`? If someone has a loader that doesn't support a file handle, they can wrap it as `lambda f: json.loads(f.read())`.